packages/store/operators/src/internals.ts

```diff
@@ -1,3 +1,12 @@
+import { StateOperator } from '@ngxs/store';
 import { RepairType } from './utils';
 
 export type Predicate<T = any> = (value?: RepairType<T> | Readonly<RepairType<T>>) => boolean;
+
+export type PatchSpec<T> = { [P in keyof T]?: T[P] | StateOperator<NonNullable<T[P]>> };
+
+export type PatchValues<T> = {
+  readonly [P in keyof T]?: T[P] extends (...args: any[]) => infer R ? R : T[P];
+};
+
+export type PatchOperator<T> = <U extends PatchValues<T>>(existing: Readonly<U>) => U;
```

Comment:
Why have these all been moved to internals?
We should keep the types with the related operator.

Reply:
Well, we have a special file for functional internals
Why keep everything in one place?

Reply:
done